src/charm.py

```diff
@@ -244,6 +245,56 @@ def _external_url_path(self) -> Optional[str]:
         # external_url.path already includes a trailing /
         return str(external_url.path) or None
 
+    @property
+    def _profiling_scrape_configs(self) -> List[Dict[str, Any]]:
+        """The scrape configuration that Parca will use for scraping profiles.
+
+        The configuration includes the targets scraped by Parca as well as Parca's
+        own workload profiles if they are not already being scraped by a remote Parca.
+        """
+        scrape_configs = self.profiling_consumer.jobs()
+        # Append parca's self scrape config if no remote parca instance is integrated over "self-profiling-endpoint"
+        if not self.self_profiling_endpoint_provider.is_ready():
+            scrape_configs.append(self._self_profiling_scrape_config)
+        return scrape_configs
+
+    @property
+    def _self_profiling_scrape_config(self) -> Dict[str, Any]:
+        """Profiling scrape config to scrape parca's own workload profiles.
+
+        This config also adds juju topology to the scraped profiles.
+        """
+        topology = JujuTopology.from_charm(self)
+
+        config = {
+            "job_name": "parca",
+            "relabel_configs": [
+                {
+                    "source_labels": [
+                        "juju_model",
+                        "juju_model_uuid",
+                        "juju_application",
+                        "juju_unit",
+                    ],
+                    "separator": "_",
+                    "target_label": "instance",
+                    "regex": "(.*)",
+                }
+            ],
+        }
+        additional_config = self._format_scrape_target(
```

Comment:
I'd refactor this to:

```
return self._format_scrape_target(...,
    job_name="parca",
    relabel_configs=[...]
)
```

I think this is better because:
- the logic of how you write a scrape target is better concentrated in _format_scrape_target
- the additional config is in fact the relabel config and the job name, not the other way around